### PR TITLE
Require Node.js 8

### PR DESCRIPTION
**Pull request by LitoMore**
### Changes

- Turn `rule.verify()` all into Promise.
- Rewrite tests with `async/await`.
- Use `Promise.all` instead of `each-async`
- Require Node.js 8

### CODE REVIEW

**Comment by LitoMore** [comment]
@SBoudrias Should we drop support for Node.js 6?

**Comment by sindresorhus** [comment]
Yes, we can drop support for Node.js 6. It will go out of LTS in two weeks.

File: lib/index.js [hunk 1]
```diff
@@ -11,22 +10,24 @@ module.exports = function () {
   console.log('\n' + chalk.underline.blue('Yeoman Doctor'));
   console.log('Running sanity checks on your system\n');
 
-  eachAsync(objectValues(rules), (rule, i, cb) => {
-    rule.verify(err => {
-      console.log((err ? symbols.error : symbols.success) + ' ' + rule.description);
+  (async () => {
+    await Promise.all(
+      objectValues(rules).map(rule => {
```
Comment (SBoudrias) [inline lib/index.js]: Can we make this into `Object.values()` now?

Comment (LitoMore) [reply lib/index.js]: @SBoudrias Sure. Code updated.

**Comment by SBoudrias** [comment]
woo thanks!

**Pull request merged**